>From 4dd0c5563d285210ca60b2cf7e9b3a4b14fbce64 Mon Sep 17 00:00:00 2001
From: Samuel Holland <samuel@sholland.org>
Date: Sun, 8 Jan 2017 19:12:08 -0600
Subject: [PATCH 1/4] Only attempt to use getcontext() on glibc

musl does not provide getcontext(). Since this function does nothing
when getcontext returns failure anyway, stub it out for non-glibc (e.g.
musl-based) environments.

Signed-off-by: Samuel Holland <samuel@sholland.org>
---
 tools/profiler/core/platform-linux.cc | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/tools/profiler/core/platform-linux.cc b/tools/profiler/core/platform-linux.cc
index 148acbc83..4908c5370 100644
--- a/tools/profiler/core/platform-linux.cc
+++ b/tools/profiler/core/platform-linux.cc
@@ -620,11 +620,13 @@ void OS::Startup() {
 void TickSample::PopulateContext(void* aContext)
 {
   MOZ_ASSERT(aContext);
+#if defined(__GLIBC__)
   ucontext_t* pContext = reinterpret_cast<ucontext_t*>(aContext);
   if (!getcontext(pContext)) {
     context = pContext;
     SetSampleContext(this, aContext);
   }
+#endif
 }
 
 void OS::SleepMicro(int microseconds)
-- 
2.11.0

